File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
from collections import Counter
def solution(nums):
    choose_cnt = len(nums)//2 # 골라야 하는 폰켓몬 개수
    cnt_nums = Counter(nums)
    #print(cnt_nums)
    
    answer = 0
    
    for key,val in cnt_nums.items():
        val-=1
        choose_cnt-=1
        #print(key,val)
        answer+=1
        if choose_cnt<=0:
            break
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        
        HashSet<Integer> hashNums = new HashSet<Integer>();
        for (int n: nums) {
            hashNums.add(n);
        }
        
        // for (int num: hashNums) {
        //     System.out.println(num);
        // }
        
        answer = Math.min(nums.length/2, hashNums.size());
        return answer;
    }
}